### PR TITLE
feat: calendar component

### PR DESCRIPTION
**Pull request by lamartire**

### CODE REVIEW

File: packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  pointer-events: none;
+}
+
+.v-date-input.theme-default .v-date-input-calendar {
+  position: absolute;
+  z-index: 1;
+  padding: 8px 4px;
+  border-radius: 4px;
+  background: #fff;
+  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.3);
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss]: maybe move shadow color to custom vars?

File: packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  transform: translateY(-50%);
+  pointer-events: none;
+}
+
+.v-date-input.theme-default .v-date-input-calendar {
+  position: absolute;
+  z-index: 1;
+  padding: 8px 4px;
+  border-radius: 4px;
+  background: #fff;
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss]: not `#fff`, but `--endpass-ui-color-white`

File: packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.3);
+}
+
+.v-date-input.theme-default .v-date-input-control {
+  width: 40px;
+  height: 40px;
+  border-width: 1px;
+  border-style: solid;
+  border-color: #fff;
+  background-color: #fff;
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss]: same with `--endpass-ui-color-white`

File: packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+.v-date-input.theme-default .v-date-input-now input {
+  flex: 0 0 auto;
+  width: 45px;
+  padding: 4px 7px 2px;
+  font-size: 14px;
+  line-height: 1.3;
+  border-radius: 4px;
+  border: none;
+  background-color: #fff;
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss]: same `--endpass-ui-color-white`

File: packages/ui/src/kit/VDateInput/VDateInput.vue [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    },
+
+    onWindowEscPress(e) {
+      if (e.key === 'Escape') {
+        this.isCalendarVisible = false;
+      }
+    },
+
+    onClickAnywhere(e) {
+      let currentNode = e.target
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.vue]: maybe we need create `outside/inside click` component?
as i see, we need not the same, but almost equal logic in `VNavSubMenu.vue`

File: packages/ui/src/kit/VDateInput/VDateInput.vue [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+        .month(innerMonth)
+        .year(innerYear)
+        .toDate();
+
+      this.$emit('change', newDate);
+      this.isCalendarVisible = false;
+    },
+
+    onWindowEscPress(e) {
+      if (e.key === 'Escape') {
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.vue]: for process esc click, please use `close-by-key-atom`

File: packages/ui/src/kit/VDateInput/VDateInput.vue [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+    },
+
+    formattedInnerMonth() {
+      return dayjs()
+        .month(this.innerMonth)
+        .format('MMMM');
+    },
+
+    currentCalendar() {
+      return getMonthCalendar(dayjs(new Date(this.innerYear, this.innerMonth)));
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.vue]: please, separate `dayjs(new Date(this.innerYear, this.innerMonth))` to single var/or vars, it's hard to read in one line

File: packages/ui/src/kit/VDateInput/VDateInput.vue [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+        v-bind="$attrs"
+        :value="formattedValue"
+        @focus="onFocusInput"
+        @blur="onBlurInput"
+      />
+      <div class="v-date-input-field-icon">
+        <v-svg-icon name="calendar" />
+      </div>
+    </div>
+    <div v-if="isCalendarVisible" class="v-date-input-calendar">
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.vue]: layout have wrong struct again ( doh...

File: packages/ui/src/kit/VDateInput/VDateInput.vue [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+      <section class="v-date-input-controls">
+        <button class="v-date-input-control" @click="onClickPreviousMonth">
+          <v-svg-icon name="chevron-left" />
+        </button>
+        <div class="v-date-input-now">
+          <span>
+            {{ formattedInnerMonth }}
+          </span>
+          <input v-model="innerYear" min="1940" type="number" />
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.vue]: why min 1940?

File: packages/ui/src/kit/VDateInput/VDateInput.vue [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+    isSelectedDay({ origin }) {
+      return (
+        this.date.date() === origin.date() &&
+        this.date.month() === origin.month() &&
+        this.date.year() === origin.year()
+      );
+    },
+
+    onBlurInput(e) {
+      console.log(e);
```
Comment (dudiq) [inline packages/ui/src/kit/VDateInput/VDateInput.vue]: `console.log` artifact, 
and can we not bind to blur/focus actions, but use css:hover?

File: packages/ui/src/utils/date.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import dayjs from 'dayjs';
+import chunk from 'lodash.chunk';
+
+/**
+ * @param {Dayjs} [target]
+ * @property {number|string} [params.year]
+ * @property {number|string} [params.month]
+ * @returns {Array<CalendarDay[]>}
+ */
+export const getMonthCalendar = (target) => {
```
Comment (dudiq) [inline packages/ui/src/utils/date.js]: if method return only last 7 days from month, maybe just rename it to `getLastDaysInMonth`?

Comment (lamartire) [reply packages/ui/src/utils/date.js]: This is not last 7 days from month, it is a full calendar monts ~4-5 weeks.

Comment (dudiq) [reply packages/ui/src/utils/date.js]: hmm, i mean, that `getMonthCalendar` name is not clear from external usage

File: packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss (hunk 1 above)
Comment (lamartire) [reply packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss]: Shadows can be different unlike colors and typography

File: packages/ui/src/kit/VDateInput/VDateInput.vue (hunk 8 above)
Comment (lamartire) [reply packages/ui/src/kit/VDateInput/VDateInput.vue]: ?

File: packages/ui/src/kit/VDateInput/VDateInput.vue (hunk 9 above)
Comment (lamartire) [reply packages/ui/src/kit/VDateInput/VDateInput.vue]: Which year in your opinion should be minimal?

File: packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss (hunk 1 above)
Comment (dudiq) [reply packages/ui/src/kit/VDateInput/VDateInput.theme-default.scss]: my fault, i started searching `box-shadow` and found, that we used `rgba(36, 42, 46, 0.1)` in most of box-shadows colors.
i mean move only `rgba(36, 42, 46, 0.1)` to vars, not this `rgba(0, 0, 0, 0.3)` color.
sorry

File: packages/ui/src/kit/VDateInput/VDateInput.vue (hunk 8 above)
Comment (dudiq) [reply packages/ui/src/kit/VDateInput/VDateInput.vue]: `class` and `v-if` must be in the next line, code style for attributes in tag fault.

File: packages/ui/src/kit/VDateInput/VDateInput.vue (hunk 9 above)
Comment (dudiq) [reply packages/ui/src/kit/VDateInput/VDateInput.vue]: it's depends maybe from prop?
but not hardcoded in html node.

File: packages/ui/src/kit/VDateInput/VDateInput.vue (hunk 8 above)
Comment (lamartire) [reply packages/ui/src/kit/VDateInput/VDateInput.vue]: Ah, it is questions to our tooling 🙃

**Pull request merged**